File: my-solutions/0861-flipping-an-image/solution.scala
object Solution {
    def flipAndInvertImage(A: Array[Array[Int]]): Array[Array[Int]] = {
        val arrayFlipped = A.map(a => a.reverse)
    val arrayInverted = arrayFlipped.map(a => a.map(el => {
      if(el == 0)
        1
      else 0
    }))
    arrayInverted
    }
}

File: my-solutions/0861-flipping-an-image/solution.py
class Solution:
    def flipAndInvertImage(self, A: List[List[int]]) -> List[List[int]]:
        for row in A:
            for i in range((len(row)+1) // 2):
                # Reverse and flip at the same time. Brilliant
                row[i], row[~i] = row[~i] ^ 1, row[i] ^ 1
        return A
